### PR TITLE
Docs: Adding a date format section

### PR DESCRIPTION
**Pull request by kirszenbaum**
This PR:
- Adds a short section on date formats (fixing #9464)
- Improves the API ref descriptions of related configuration options: `dateFormat` and `correctFormat`

[skip changelog]

### CODE REVIEW

File: handsontable/src/dataMap/metaManager/metaSchema.js [hunk 1]
```diff
@@ -1484,12 +1482,11 @@ export default () => {
...
     /**
-     * The `dateFormat` option configures [`date`](@/guides/cell-types/date-cell-type.md) cells' date format.
+     * The `dateFormat` option configures the date format accepted by [`date`](@/guides/cell-types/date-cell-type.md) cells.
      *
-     * You can set the `dateFormat` option to a date format string. The default value is: `'DD/MM/YYYY'`.
+     * You can set the `dateFormat` option to a string with a proper date format. The default value is: `'DD/MM/YYYY'`.
      *
-     * To enforce the date format set by the `dateFormat` option,
-     * use the [`correctFormat`](#correctFormat) option.
+     * To correct dates whose format doesn't match the `dateFormat` setting, use the [`correctFormat`](#correctFormat) option.
```
Comment (kirszenbaum) [inline handsontable/src/dataMap/metaManager/metaSchema.js:1489]:
```suggestion
     * To automatically correct dates whose format doesn't match the `dateFormat` setting, use the [`correctFormat`](#correctFormat) option.
```

File: docs/next/guides/cell-types/date-cell-type.md [hunk 2]
```diff
@@ -24,7 +24,38 @@ Note that date cell requires additional modules :
...
+
+### Changing the date format
+
+To change the date format accepted by `date` cells, set the [`dateFormat`](@/api/options.md#dateformat) configuration option to a string with your preferred format. For example:
+
+```js
+dateFormat: 'YYYY-MM-DD',
+```
+
+### Correcting invalid dates
```
Comment (kirszenbaum) [inline docs/next/guides/cell-types/date-cell-type.md:43]:
```suggestion
### Autocorrecting invalid dates
```

**Review by AMBudnik — DISMISSED** [review]
@kirszenbaum approved. It's so much better now. Thanks!

File: docs/next/guides/cell-types/date-cell-type.md [hunk 3]
```diff
@@ -24,7 +24,38 @@ Note that date cell requires additional modules :
...
+
+```js
+dateFormat: 'YYYY-MM-DD',
+```
+
+### Autocorrecting invalid dates
+
+By default, when the user enters a date in a format that doesn't match the [`dateFormat`](@/api/options.md#dateformat) setting, the date is treated as invalid.
+
+You can set Handsontable to format such dates automatically, so they match the required format. To do this, set the [`correctFormat`](@/api/options.md#correctformat) option to `true`. For example:
```
Comment (kirszenbaum) [inline docs/next/guides/cell-types/date-cell-type.md:47]:
```suggestion
You can let Handsontable correct such dates automatically, so they match the required format. To do this, set the [`correctFormat`](@/api/options.md#correctformat) option to `true`. For example:
```

File: docs/next/guides/cell-types/date-cell-type.md [hunk 4]
```diff
@@ -24,7 +24,38 @@ Note that date cell requires additional modules :
 <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/pikaday@1.8.2/css/pikaday.css">
 ```
 
-All data entered into the date-type cells are validated against the default date format `DD/MM/YYYY` unless another format is provided. If you enable the `correctFormat` config item, the dates will be automatically formatted to match the desired format.
+## Date format
+
+`date` cells accept strings, formatted in line with the [`dateFormat`](@/api/options.md#dateformat) setting.
```
Comment (kirszenbaum) [inline docs/next/guides/cell-types/date-cell-type.md:29]:
```suggestion
`date` cells accept strings that are formatted in line with the [`dateFormat`](@/api/options.md#dateformat) setting.
```

**Review by AMBudnik — APPROVED** [review]

**Pull request merged**